### PR TITLE
win32com: server: Adding reg_server_cmdline for registration

### PR DESCRIPTION
**Pull request by thopiekar**
Not every application is able to handle foreslash. Also there might be cases where the option "Automate" is already taken. Therefore I added the possibility to let the user define her/his own option.
I don't know in which cases it might be benefitial, but I left the door open for projects, which provide pure servers. These won't need an option at all. These projects only have to provide an empty string.
The value `None` will lead to the default value '/Automate'.

### CODE REVIEW

File: com/win32com/server/register.py [hunk 1]
```diff
@@ -172,6 +173,8 @@ def RegisterServer(clsid,
      clsctx -- One of the CLSCTX_* constants.
      addnPath -- An additional path the COM framework will add to sys.path
                  before attempting to create the object.
+     exeOption -- Option under which the server shall be started. By detault
+                  this varaible is set to '/Automate'.
```
Comment (mhammond) [inline com/win32com/server/register.py]: typo in "varaible"

**Comment by thopiekar** [comment]
Hey @mhammond ,

the motivation on this is because of leaking support for Python3.5 by Py2Exe. Cura is needs because of special CI requirements Python3.5 and according to my researches Python3.5 also comes with significant changes, so anything newer then Python 3.4 does not work.
Therefore Cura is using cx_Freeze now to create executables.

In the beginning I didn't know that PyWin32 is registering the executable by default with this option, so I was wondering why my application as always dying when dispatching it.
After a while I noticed there is this convention, which is not understood by cx_Freeze.

However, I created this code during this time and can imaging that for some people it might be useful (even only for development purposes) to override this value.
Also I'm interested in what has been done inside py2exe. As I'm still struggling getting events working, I think I still missed something in py2exe. But, however, this is a different story.

File: com/win32com/server/register.py (hunk 1 above)
Comment (thopiekar) [reply com/win32com/server/register.py]: Fixed in https://github.com/mhammond/pywin32/pull/1130/commits/8f287ff5919a98edc1cacd58111aa30c661256a5

**Comment by mhammond** [comment]
Do you intend pushing a new version with the requested changes?

**Comment by mhammond** [comment]
It's been a couple of months since changes were requested, so I'll close this, but we can reopen it if the changes arrive.

**Comment by thopiekar** [comment]
Hey @mhammond! 😄 

Is there any chance to continue on this? I came back to my project, which needs this PR.
I would be glad if I could rework this and get it merged finally. 😉 

Regards!
- Thomas

**Comment by mhammond** [comment]
sure!

**Comment by thopiekar** [comment]
Great, thanks!

Please confirm whether I understand your last remarks correctly:
 * Your main intention is to combine the path to executable and option variable to one cmdline variable, which overwrites everything, right?
* If this cmdline variable is used, the overwrite should also work, when we are not frozen, correct? Personally, I just hit a situation right now that I need to run script non-frozen, but still want to overwrite. So that would be even useful for me.

Another remark from myself, while reviewing the whole module:
* I also need to update the "RegisterClasses" function. That one needs to extract this option and pass it to RegisterServer.

**Comment by mhammond** [comment]
Sorry I didn't reply here! Are you still interested in following up here? In case you are:

>     * Your main intention is to combine the path to executable and option variable to one cmdline variable, which overwrites everything, right?

Yeah, that makes sense to me. If `reg_server_cmdline` is None, it works exactly as it does now, including a different cmdline whether frozen or not. If it's not None, it's used unconditionally.

>     * If this cmdline variable is used, the overwrite should also work, when we are not frozen, correct? Personally, I just hit a situation right now that I need to run script non-frozen, but still want to overwrite. So that would be even useful for me.

Yep

> Another remark from myself, while reviewing the whole module:
> 
>     * I also need to update the "RegisterClasses" function. That one needs to extract this option and pass it to RegisterServer.

Yep! `reg_server_cmdline` would actually be the class attribute. The function param could be named, say, `serverCmdline` or similar.

**Comment by niteshJyo** [comment]
How to import pywin32 latest version in Python

**Comment by mhammond** [comment]
Thanks for this, but it appears to have stalled - feel free to reopen if you pick this back up in the future.

**Pull request closed without merging**